driver-embedded-android/src/androidTest/java/com/mongodb/embedded/client/Fixture.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.embedded.client;
+
+
+import android.content.Context;
+import android.support.test.InstrumentationRegistry;
+
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.connection.ServerVersion;
+
+import org.bson.BsonDocument;
+import org.bson.BsonValue;
+import org.bson.Document;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.util.Arrays.asList;
+
+
+/**
+ * Helper class for the acceptance tests.
+ */
+public final class Fixture {
+    private static final String APPLICATION_NAME = "testApp";
+    private static final String DATA_DIR = "dataDir";
+    private static final String DEFAULT_DATABASE_NAME = "JavaDriverTest";
+
+    private static MongoClient mongoClient;
+    private static MongoEmbeddedSettings mongoEmbeddedSettings;
+    private static MongoClientSettings mongoClientSettings;
+    private static ServerVersion serverVersion;
+
+    static synchronized MongoClient getMongoClient() {
+        if (mongoClient == null) {
+            cleanDBPath();
+            MongoClients.init(getMongoEmbeddedSettings());
+            mongoClient = MongoClients.create(getMongoClientSettings());
+            Runtime.getRuntime().addShutdownHook(new ShutdownHook());
+        }
+        return mongoClient;
+    }
+
+    static synchronized MongoClientSettings getMongoClientSettings() {
+        if (mongoClientSettings == null) {
+            File dataDir = InstrumentationRegistry.getContext().getDir(DATA_DIR, Context.MODE_WORLD_WRITEABLE);
```

Comment:
Getting the dataDir via the android context was an early fix for getting this to work.

Reply:
And still necessary?

Reply:
Yes, we need the settings to use the correct file path that it can access. Otherwise it segfaults in the testing environment.